### PR TITLE
fix(eslint): manual fix of eslint errors on package schematics

### PR DESCRIPTION
**Pull request by matthieu-crouzet**

### CODE REVIEW

File: packages/@o3r/schematics/src/utility/collection.spec.ts [hunk 1]
```diff
@@ -43,18 +42,18 @@ const angularJsonNoGeneric: WorkspaceSchema = {
...
-describe('Get schematics options', () => {
-  const createFakeContext = (collection: string, name: string): any => ({
-    schematic: {
-      description: {
-        collection: {
-          name: collection
-        },
-        name
-      }
+const createFakeContext = (collection: string, name: string): any => ({
```
Comment (vscaiceanu-1a) [inline packages/@o3r/schematics/src/utility/collection.spec.ts:45]: wasn't this rule disabled for tests? (I am ok with the change, just curious)

File: packages/@o3r/schematics/src/utility/package-version.ts [hunk 2]
```diff
@@ -28,14 +28,13 @@ export function getPackageVersion(packageJsonPath: string) {
...
 export const findClosestPackageJson = (currentPath: string, tree?: Tree): string | undefined => {
-  const dir = (tree ? posix.dirname : dirname)(currentPath);
+  const dir = (tree ? posix.dirname.bind(posix) : dirname)(currentPath);
   if (dir === currentPath || currentPath === '/') {
     return undefined;
   }
 
-  const packageJsonPath = (tree ? posix.join : join)(dir, 'package.json');
-  return (tree ? tree.exists : existsSync)(packageJsonPath) ? packageJsonPath : findClosestPackageJson(dir, tree);
+  const packageJsonPath = (tree ? posix.join.bind(posix) : join)(dir, 'package.json');
```
Comment (vscaiceanu-1a) [inline packages/@o3r/schematics/src/utility/package-version.ts:38]: from which rule is this coming?

**Review by fpaul-1A — APPROVED** [review]

File: packages/@o3r/schematics/src/utility/collection.spec.ts (hunk 1 above)
Comment (matthieu-crouzet) [reply packages/@o3r/schematics/src/utility/collection.spec.ts:45]: We said that we will deactivate this rule after the migration, for now we try to fix a maximum when possible and meaningful

File: packages/@o3r/schematics/src/utility/package-version.ts (hunk 2 above)
Comment (matthieu-crouzet) [reply packages/@o3r/schematics/src/utility/package-version.ts:38]: @typescript-eslint/unbound-method

**Review by vscaiceanu-1a — APPROVED** [review]

**Pull request merged**